modules/virt-configuring-vm-real-time.adoc

```diff
@@ -49,56 +49,127 @@ spec:
 +
 [source,terminal]
 ----
-$ dnf install -y tuned-profiles-cpu-partitioning
+$ dnf install kernel-rt tuned cloud-init realtime-tests
 ----
+.. To achieve low-latency tuning by using the `realtime-virtual-guest` profile in the TuneD application, run the following commands:
 +
 [source,terminal]
 ----
-$ echo isolated_cores=2-9 > /etc/tuned/cpu-partitioning-variables.conf
+$ dnf --enablerepo=rt install -y tuned-profiles-realtime
 ----
-The first two CPUs (0 and 1) are set aside for house keeping tasks and the rest are isolated for the real-time application.
 +
 [source,terminal]
 ----
-$ tuned-adm profile cpu-partitioning
+$ dnf --enablerepo=nfv install -y tuned-profiles-nfv-guest
 ----
+.. Edit `/etc/tuned/realtime-virtual-guest-variables.conf`:
++
+[source,conf]
+----
+#
+# Variable settings below override the definitions from the
+# /etc/tuned/realtime-variables.conf file.
+
+#
+# Core isolation
+#
+# The 'isolated_cores=' variable below controls which cores should be
+# isolated. By default we reserve 1 core per socket for housekeeping
+# and isolate the rest. But you can isolate any range as shown in the
+# examples below. Just remember to keep only one isolated_cores= line.
+#
+# Examples:
+# isolated_cores=2,4-7
+# isolated_cores=2-23
+#
+# Reserve 1 core per socket for housekeeping, isolate the rest.
+# Change this for a core list or range as shown above.
+isolated_cores=2-3 <1>
+
+#
+# Uncomment the 'isolate_managed_irq=Y' bellow if you want to move kernel
+# managed IRQs out of isolated cores. Note that this requires kernel
+# support. Please only specify this parameter if you are sure that the
+# kernel supports it.
+#
+isolate_managed_irq=Y <2>
 
+#
+# Set the desired combined queue count value using the parameter provided
+# below. Ideally this should be set to the number of housekeeping CPUs i.e.,
+# in the example given below it is assumed that the system has 4 housekeeping
+# (non-isolated) CPUs.
+#
+# netdev_queue_count=4
+----
+<1> The first two CPUs (0 and 1) are set aside for house keeping tasks and the rest are isolated for the real-time application.
+<2> isolate_managed_irq should be set to `Y`
+.. Activate the TuneD profile:
++
+[source,terminal]
+----
+$ tuned-adm profile realtime-virtual-guest
+----
+.. Set the real-time kernel as the default by using the GRUB boot loader command-line interface:
++
+[source,terminal]
+----
+$ grubby --set-default=<installed_rt_kernel_version>
+----
+.. Set the kernel arguments using the GRUB boot loader command-line interface:
++
+[source,terminal]
+----
+$ grubby --args="iommu=pt intel_iommu=on default_hugepagesz=1G idle=poll" --update-kernel=$(grubby --default-kernel)
+----
 . Restart the VM to apply the changes.
 
+.Verification
+* Use the `cyclictest` tool to verify that the real-time guest is configured properly:
++
+[source,terminal]
+----
+$ cyclictest --priority 1 --policy fifo -h 50 -a 2-3 --mainaffinity 0,1 -t 2 -m -q -i 200 -D 12h
```

Comment:
1. The user should be alerted that the `-D` parameter determines the test duration, so if they use this example it means a 12h test.
2. This also requires root

Reply:
@sjhala-ccs can we somehow refer to the man page of cyclictest?

Reply:
> The user should be alerted that the -D parameter determines the test duration, so if they use this example it means a 12h test.

This should still be addressed.